HIP/hip-513.md

```diff
@@ -0,0 +1,293 @@
+---
+hip: 513
+title: Smart Contract Traceability Extension
+author: Stoyan Panayotov <stoyan.panayotov@limechain.tech>, Mustafa Uzun <mustafa.uzun@limechain.tech>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft
+last-call-date-time:
+created: 2022-06-08
+discussions-to:
+updated:
+requires: 206, 435, 260
+---
+
+## Abstract
+
+This HIP overrides [HIP-260](./hip-260.md) by moving traceability info from `ContractFunctionResult` to sidecar.
+
+Extends traceability with new sidecar records `ContractAction` and `ContractBytecode`.
+
+Proposes additional data placed in sidecar transaction records to enable full
+replay of transactions by mirror nodes and downstream users of Hedera.
+
+## Motivation
+
+Enabling mirror nodes to download traceability info only if they need or want to process it.
+
+Enabling full replay of smart contract service transactions will improve the
```

Comment:
The flow will also support estimation of execution or values like gas to be consumed.
So it's worth noting that.

Reply:
DONE